### ISSUE TITLE
Hometasks

### ISSUE THREAD

**Issue by ZhuChara2004** [MastersAcademy/Programming-Basics#1]
@Тут будуть додаватися домашні завдання. Для швидких обговорень є Gitter. Коментарі, в яких немає сенсу будуть видалятись. Домашні завдання виконувати в каталозі homeworks. Кожне завдання у власному каталозі з назвою типу Homework X де X - номер завдання.

### Hometask 00
- **Уважно прочитати і усвідомити README в корні репозиторію**
- Форкнути репозиторій. У форкнутому репозиторії в каталозі homeworks створити каталог для домашніх завдань. Всередині створити каталог Homework-0 і додати в нього файл README.md з будь-яким вмістом. Зробити пул реквест в основний репозиторій.
- **обов'язково** пройти туторіал [Try GitHub](https://try.github.io). Створувати файли через інтерфейс гітхаба - неправильно. Та і доданий цей функціонал місяці чотири тому. Все повинно робитися на вашому комп'ютері і заливатись в репозиторій за допомогою пушів.

### Hometask 01
- Подивитись скрінкаст сесії 1
- Написати додаток, який буде запитувати в користувача його анкетні дані і виводити їх в консоль
- _Додатково_: зберегти запитані дані в файл в довільному форматі

### Hometask 02
- Доповнити попередню роботу колекціями

### Hometask 03
- Створити markdown([файли з розширенням .md](https://habrahabr.ru/sandbox/23123/) ) файл, в якому, на прикладi об'єкта рельного свiту, описати основнi поняття ООП.(клас, об'єкт, спадковість, поіморфізм, інкапсуляція). Також застосувати(всі або частково) принципи SOLID.
[принципи SOILD](https://web-creator.ru/articles/solid)

### Hometask 04
- Доповнити попереднє завдання прикладами Control Flow

### Hometask 05
- На основі вашого творчого завдання написати додаток з класами
  Мають використовуватися:
  - наслідування
  - інкапсуляція

### Hometask 06
1.Прочитати наступнi статтi:
  http://procomputer.su/comp-gramotnost/84-ip-adres-chto-eto-takoe
  http://ua.smart-ip.net/what-is-a-port
  http://www.4stud.info/networking/lecture5.html
  http://bourabai.kz/dbt/client1.htm
  https://habrahabr.ru/post/215117/

2. Створити в вашому додатку функцію яка буде брати обьект вашого класу i представляти його в JSON та XML форматах, зберігаючи йих в видповідних JSON та XML файлах.

**Comment by SpraTV** [comment]
О_О, в мене таке враження, що до 5 листопада я тільки читатиму інструкцію до github.

**Comment by ssemenukha** [comment]
Така у нас задача черпати невичерпне :) Хочеться вивчити - вчімо.
Я теж раніше не юзав гітхаб. Нічого, вїдемо :))

Не поганий ГУЙ для ГІТа https://www.gitkraken.com/

**Comment by mr-valentos** [comment]
Чому?

2016-10-28 15:23 GMT+03:00 SpraTV <notifications@github.com>:

**Comment by SpraTV** [comment]
слаба англійська, + мало базових знань програмування, а в результаті не знання програмістських сленгів, що значно ускладнює сприйняття задачі, а відповідно і її виконання. Це не те що я жаліюсь, просто пояснив ситуацію))  я справлюсь, думаю тільки більше часу на це піде))

**Comment by mr-valentos** [comment]
У меня примерно та же ситуация) Но github вроде как легкий в освоении, если
что спрашивай что не понятно😉

2016-10-28 15:32 GMT+03:00 SpraTV <notifications@github.com>:

**Comment by SpraTV** [comment]
ок, дякую

**Comment by keyhit** [comment]
Привіт! Хочу уточнити. Шлях до домашньої роботи повинен мати такий вигляд " homeworks/логін гітхабу/homework 0/README.md"? Дякую.

**Comment by ancoooo** [comment]
а як створити каталог в репозиторії?

**Comment by mr-valentos** [comment]
create new file и потом просто прописываешь его через "/"

2016-10-28 16:00 GMT+03:00 ancoooo <notifications@github.com>:

**Comment by ancoooo** [comment]
а что прописивать?

**Comment by mr-valentos** [comment]
Просто название

2016-10-28 16:07 GMT+03:00 ancoooo <notifications@github.com>:

**Comment by mr-valentos** [comment]
Вроде как путь к файлу прописываешь в окне Windows

2016-10-28 16:09 GMT+03:00 Valentin Nik <nerpavlad@gmail.com>:

**Comment by mr-valentos** [comment]
Вроде да, только просили не только логин (name.surname_githubUsername)

2016-10-28 15:49 GMT+03:00 keyhit <notifications@github.com>:

**Comment by keyhit** [comment]
Дякую!

**Comment by SpraTV** [comment]
а що трапилось??, мабуть десь помилка, якщо файл закритий??

**Comment by Elmarfdk** [comment]
Так, насколько я понял. Сначала кликаем на fork справа вверху, создается своя копия репозитория. Далее Create a new file в названии прописываем homeworks/name.surname_githubUsername/Homework 0/README.md. Потом жмем Pull requests и ждем пока нашу папку с файлом создаст админ. Правильно не?

**Comment by Tima3** [comment]
Я так делал ! Но почему-то у меня имя файла README.md ,  что-то я не так делаю )

**Comment by Elmarfdk** [comment]
Если вы успели испортить форкнутый репозиторий, заходим в него и во вкладке settings внизу удаляем его.

**Comment by Tima3** [comment]
меня уже внесли в список ?

**Comment by Vasyl-Shche** [comment]
А що то за дивний каталог в корні репозиторія?

**Comment by Tima3** [comment]
я некогда не юзал git , помогите )

**Comment by Vasyl-Shche** [comment]
Заходиш в репозиторію тиснеш create new file потім вказуєш путь типу /homeworks/Vasyl.Shcherbatenko_Vasyl-Shche/Homework-0/README.md де Vasyl-name ,Shcherbatenko-surname i Vasyl-Shche-логин github

**Comment by ZhuChara2004** [comment]
До кінця наступної середи (16.11.2016) поправити назви каталогів для домашніх завдань. Неправильно названі каталоги в четвер будуть видалені.

**Comment by Nako68l** [comment]
Правильное название папки должно иметь никнейм в соответствии с гитхабом или тоже переносить всё под один регистр?

**Comment by nomatterhowyoutry** [comment]
А как понять, что у имя моего каталога неправильное?

**Comment by Isildy** [comment]
Вcем привет в связи определенными обстоятельствами навсякий случай уточняю время дедлайна домашних работ.  ДЗ нужно сдать **ДО 18:30  16.11.2016**. После этого срока ДЗ приниматся не будут. И по традиции поставте confused в коментари, те кто прочитал.

**Comment by darkwoolf10** [comment]
Назви каталогів це ж по типу pasha.fudko_darkwoolf10, норм?Чи ще може якись э шаблон для папок в них?

**Comment by alextriam** [comment]
имя-фамилие должны быть в нижнем регистре, после _ пишется логин на гите как есть.

**Comment by Nako68l** [comment]
в шаблоне ник с маленькой, когда у Дмитрия с большой, так что предполагаю нужно всё под нижним регистром

**Comment by Isildy** [comment]
Место проведеия и доп информация касающееся следующего занятия https://github.com/MastersAcademy/Programming-Basics/issues/304

К следующему занятию **обязательно** ознакомится с основными понятиями Обьектно Ориентированного Програмирования(ООП). Такими как: 
- класc
- объект
- абстракция
- наследование
- полиморфизм
- инкапсуляция

 Ожидается понимание что это такое и как может применятся (не обязательно в програмировании можно на примерах реального мира).

**Comment by Isildy** [comment]
### Hometask 03
- Створити markdown([файли з розширенням .md](https://habrahabr.ru/sandbox/23123/) файл, в якому, на прикладi об'єкта рельного свiту, описати основнi поняття ООП.(клас, об'єкт, спадковість, поіморфізм, інкапсуляція). Також застосувати(всі або частково) принципи SOLID.

[принципи SOILD](https://web-creator.ru/articles/solid)

Завдання необхідно виконати **до 23:59 29.11.2016**

**Comment by Isildy** [comment]
###Hometask 06

1.Прочитати наступнi статтi:
  http://procomputer.su/comp-gramotnost/84-ip-adres-chto-eto-takoe
  http://ua.smart-ip.net/what-is-a-port
  http://www.4stud.info/networking/lecture5.html
  http://bourabai.kz/dbt/client1.htm
  https://habrahabr.ru/post/215117/

2. Створити в вашому додатку функцію яка буде брати обьект вашого класу i представляти його в JSON та XML форматах, зберігаючи йих в видповідних JSON та XML файлах.

**Comment by Isildy** [comment]
Дедлайн до 18.01.2017 до 18:00

**Comment by doctorrrrr** [comment]
https://web-creator.ru/articles/solid
Хлопці, переробіть або закрийте. Не потрібно нам посилання на країну Пусте місце.
